### DISCUSSION TITLE
What should I put in a profile for a program to ensure it can't have keylogging capabilities?

### DISCUSSION THREAD

**Discussion by fury182** [netblue30/firejail#5075] [Q&A] [answered]
What should I put in a profile for a program to ensure it can't have keylogging capabilities?

I'm on standard Ubuntu 20.04 LTS.

Thanks

Relates to:

* #5076

**Comment by ghost** [comment, marked as answer]
Very good question. Also a rather complex topic, due to it being related running Xorg, Xwayland or full Wayland. I think your other question about [clipboard protection](https://github.com/netblue30/firejail/discussions/#5076) falls under the same category. There isn't a `set-and-forget` option AFAIK. You can read up on the topic in some of our issues:

- https://github.com/netblue30/firejail/issues/875
- https://github.com/netblue30/firejail/issues/998
- https://github.com/netblue30/firejail/issues/2248

For a how-to guide IMO https://firejail.wordpress.com/documentation-2/x11-guide/ is still your best bet.
